PopupBridge/src/main/java/com/braintreepayments/api/PopupBridgeLifecycleObserver.java

```diff
@@ -0,0 +1,76 @@
+package com.braintreepayments.api;
+
+import static androidx.lifecycle.Lifecycle.Event.ON_RESUME;
+
+import android.os.Handler;
+import android.os.Looper;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.VisibleForTesting;
+import androidx.fragment.app.FragmentActivity;
+import androidx.lifecycle.Lifecycle;
+import androidx.lifecycle.LifecycleEventObserver;
+import androidx.lifecycle.LifecycleOwner;
+
+import java.lang.ref.WeakReference;
+
+class PopupBridgeLifecycleObserver implements LifecycleEventObserver {
+
+    @VisibleForTesting
+    final PopupBridgeClient popupBridgeClient;
+
+    PopupBridgeLifecycleObserver(PopupBridgeClient popupBridgeClient) {
+        this.popupBridgeClient = popupBridgeClient;
+    }
+
+    @Override
+    public void onStateChanged(@NonNull LifecycleOwner lifecycleOwner, @NonNull Lifecycle.Event event) {
```

Comment:
Will we add docstrings / update the README with how merchants should use this?

Reply:
@scannillo in `java` the default access modifier `package-private` so `PopupBridgeLifeCycleObserver` is only visible to clases that are on the same package. This class is an internal class to observe the context lifecycle so that the `PopupBridgeClient` can retrieve a `BrowserSwitchResult` effectevily

Reply:
Ah, thats for clarifying. I think the `public` on line 27 threw me off. Should that be changed?

Reply:
No problem! 
No, because that is the signature of method in `LifecycleEventObserver` interface, that `PopupBridgeLifecycleObserver` implements.
More context of what is going on:
Because this does a browser switch, we need to know when the Merchants activity is back at the front, in order to do so, we register an observer to its lifecycle. Then Android informs our observer of any state change in the activity through this method. In line 28 we want to know if the state is `ON_RESUME`, meaning if the activity is at the front of the app.

Reply:
Here it actually does need to be public since it's an `@Override`. Like Jose mentioned though the class will be hidden from merchants so they won't have access to it.

Reply:
Awesome, thank you both for the explanation!